src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/ProjectImports/ProjectImportsCommandGroupHandlerBase.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Runtime.ExceptionServices;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Tree.ProjectImports;
+
+/// <summary>
+/// Handles opening of files displayed in the project imports tree.
+/// </summary>
+internal abstract class ProjectImportsCommandGroupHandlerBase : IAsyncCommandGroupHandler
+{
+    protected ProjectImportsCommandGroupHandlerBase()
+    {
+    }
+
+    protected abstract bool IsOpenCommand(long commandId);
+
+    protected abstract bool IsOpenWithCommand(long commandId);
+
+    protected abstract void OpenItems(long commandId, IImmutableSet<IProjectTree> items);
+
+    public Task<CommandStatusResult> GetCommandStatusAsync(IImmutableSet<IProjectTree> items, long commandId, bool focused, string? commandText, CommandStatus progressiveStatus)
```

Comment:
While these aren't VS-specific types, this is still tied to how VS operates. `commandId` will always be a VS-specific value, and `CommandStatus` is just an enum that wraps OLECMDF values. CPS doesn't really have a concept of commands or command status separate from the VS commands system.

Reply:
Got it. 

I thought this was a general concept.